File: java/src/test/java/com/alxarafe/domain/rules/valueobject/MovementTypeTest.java
package com.alxarafe.domain.rules.valueobject;

import com.alxarafe.app.domain.rules.valueobject.MovementType;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MovementTypeTest {

    @Test
    void values() {
        assertEquals(MovementType.INBOUND, MovementType.valueOf("INBOUND"));
        assertEquals(MovementType.OUTBOUND, MovementType.valueOf("OUTBOUND"));
        assertEquals(MovementType.TRANSFER, MovementType.valueOf("TRANSFER"));
        assertEquals(MovementType.ADJUSTMENT, MovementType.valueOf("ADJUSTMENT"));
        assertEquals(4, MovementType.values().length);
    }
}

File: php/tests/Unit/Domain/Rules/ValueObject/MovementTypeTest.php
<?php

declare(strict_types=1);

namespace Tests\Unit\Domain\Rules\ValueObject;

use Alxarafe\App\Domain\Rules\ValueObject\MovementType;
use PHPUnit\Framework\TestCase;

final class MovementTypeTest extends TestCase
{
    public function testValues(): void
    {
        self::assertSame('INBOUND', MovementType::INBOUND->value);
        self::assertSame('OUTBOUND', MovementType::OUTBOUND->value);
        self::assertSame('TRANSFER', MovementType::TRANSFER->value);
        self::assertSame('ADJUSTMENT', MovementType::ADJUSTMENT->value);
    }
}
